### PR TITLE
add type for prim with-handlers*

### LINKED ISSUE #896: with-handlers* couldn't be used in typed/racket
### What version of Racket are you using?
7.5
### What program did you run?
(with-handlers* ([exn:fail? (λ (e) 1)])
  2)
### What should have happened?
2
### If you got an error message, please include it here.
Type Checker: missing type for identifier;
 consider using `require/typed' to import it
  identifier: break-enabled-key
  from module: (quote #%paramz) in: (with-handlers* ((exn:fail? (λ (e) 1))) 2)

### PR DESCRIPTION
**Pull request by Syntacticlosure**
fix #896

### CODE REVIEW

**Comment by samth** [comment]
This needs a fix in typed/scheme/base.rkt, which should probably be pretty obvious when you look at the file.

File: typed-racket-test/succeed/with-handlers-star.rkt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#lang typed/racket/base
+(with-handlers* ([exn:fail? (λ (e) 2)])
+  1)
+(with-handlers* ([exn:fail? (λ (e) 2)])
```
Comment (samth) [inline typed-racket-test/succeed/with-handlers-star.rkt]: Somehow this is causing the integration test suite to timeout.

Comment (Syntacticlosure) [reply typed-racket-test/succeed/with-handlers-star.rkt]: It was caused by a typo : 
```racket
(define-syntax with-handlers*: (make-with-handlers #'with-handlers*:))
```

**Comment by samth** [comment]
It would be good to add documentation (preferably with a `history` note) to `typed-racket-doc/typed-racket/scribblings/reference/special-forms.scrbl` following the pattern for `with-handlers`. Do you want to do that?

File: typed-racket-doc/typed-racket/scribblings/reference/special-forms.scrbl [hunk 2]
```diff
@@ -727,6 +728,11 @@ Identical to @|with-handlers-id| from @racketmodname[racket/base]
 but provides additional annotations to assist the typechecker.
 }
 
+@defidform[with-handlers*]{
+Identical to @|with-handlers*-id| from @racketmodname[racket/base]
+but provides additional annotations to assist the typechecker.
```
Comment (samth) [inline typed-racket-doc/typed-racket/scribblings/reference/special-forms.scrbl]: Can you add a `@history` note here?

**Pull request merged**